File: 351002/Yakhnovets/rest-task/src/reaction/controller/reaction.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  Param,
  ParseIntPipe,
  Post,
  Put,
} from '@nestjs/common';
import { ReactionService } from '../service/reaction.service';
import { ReactionRequestTo } from '../dto/reaction-request.to';
import { ReactionResponseTo } from '../dto/reaction-response.to';

@Controller('api/v1.0/reactions')
export class ReactionController {
  constructor(private readonly service: ReactionService) {}

  @Get()
  getAll(): Promise<ReactionResponseTo[]> {
    return this.service.getAll();
  }

  @Get(':id')
  getById(@Param('id', ParseIntPipe) id: number): Promise<ReactionResponseTo> {
    return this.service.getById(id);
  }

  @Post()
  create(@Body() dto: ReactionRequestTo): Promise<ReactionResponseTo> {
    return this.service.create(dto);
  }

  @Put(':id')
  update(
    @Param('id', ParseIntPipe) id: number,
    @Body() dto: ReactionRequestTo,
  ): Promise<ReactionResponseTo> {
    return this.service.update(id, dto);
  }

  @Delete(':id')
  @HttpCode(204)
  delete(@Param('id', ParseIntPipe) id: number): Promise<void> {
    return this.service.delete(id);
  }
}

File: 351003/Razhnou/task361/discussion/src/main/java/by/bsuir/task361/discussion/controller/ReactionController.java
package by.bsuir.task361.discussion.controller;

import by.bsuir.task361.discussion.dto.request.ReactionRequestTo;
import by.bsuir.task361.discussion.dto.response.ReactionResponseTo;
import by.bsuir.task361.discussion.service.ReactionService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/reactions")
public class ReactionController {
    private final ReactionService reactionService;

    public ReactionController(ReactionService reactionService) {
        this.reactionService = reactionService;
    }

    @PostMapping
    public ResponseEntity<ReactionResponseTo> create(@Valid @RequestBody ReactionRequestTo request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(reactionService.createFromRest(request));
    }

    @GetMapping
    public List<ReactionResponseTo> findAll() {
        return reactionService.findAll();
    }

    @GetMapping("/{id}")
    public ReactionResponseTo findById(@PathVariable Long id) {
        return reactionService.findById(id);
    }

    @PutMapping
    public ReactionResponseTo update(@Valid @RequestBody ReactionRequestTo request) {
        return reactionService.update(request);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        reactionService.delete(id);
        return ResponseEntity.noContent().build();
    }
}
